transformer/metadata.go

```diff
@@ -79,12 +79,20 @@ func (meta *Metadata) Transform(c *skogul.Container) error {
 }
 
 // flattenStructure copies a nested object/array to the root level
-func flattenStructure(nestedPath []string, metric *skogul.Metric) error {
-	newPath := nestedPath[0]
+func flattenStructure(nestedPath []string, separator string, metric *skogul.Metric) error {
+	nestedObjectPath := nestedPath[0]
+	var pathFmt string
+	if separator == "drop" {
+		pathFmt = "%s%s"
+		nestedObjectPath = ""
+	} else {
+		pathFmt = "%s" + separator + "%s"
+	}
 
-	if len(nestedPath) > 1 {
+	// Create a nested path unless configuration says not to
+	if separator != "drop" && len(nestedPath) > 1 {
 		for _, p := range nestedPath[1:] {
-			newPath = fmt.Sprintf("%s__%s", newPath, p)
+			nestedObjectPath = fmt.Sprintf(pathFmt, nestedObjectPath, p)
```

Comment:
Clearly this works, but wouldn't it be cleaner to just use Sprintf("%s%s%s","nestedObjectPath, separator, p) ?

Granted, it doesn't give us _much_, though...

Hmm, what we could do, is make the Data.FlattenSeparator a string *pointer*, instead of a string. It's a bit... weird... but it will allow us to distinguish between a config with a blank separator (valid pointer to an empty string) and a config with an undefined separator (nil). But, for now, I think using "drop" is fine - providing a mechanism for default values for strings is something that belongs in the config parser, not a random module.